constellations/orion/src/controllers/classes_by_semesters_controller_test.go

```diff
@@ -0,0 +1,223 @@
+package controllers_test
+
+import (
+	"encoding/json"
+	"errors"
+	"net/http"
+	"testing"
+
+	"github.com/ahsu1230/mathnavigatorSite/constellations/orion/src/controllers/testUtils"
+	"github.com/ahsu1230/mathnavigatorSite/constellations/orion/src/domains"
+	"github.com/ahsu1230/mathnavigatorSite/constellations/orion/src/repos"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestOneSemesterOneProgramOneClass_Success(t *testing.T) {
+	// Mock 1 program, 1 semester, 1 class
+	testUtils.ProgramRepo.MockSelectAll = func(publishedOnly bool) ([]domains.Program, error) {
+		return createMockPrograms(1), nil
+	}
+	repos.ProgramRepo = &testUtils.ProgramRepo
+
+	testUtils.SemesterRepo.MockSelectAll = func(publishedOnly bool) ([]domains.Semester, error) {
+		return createMockSemesters(1), nil
+	}
+	repos.SemesterRepo = &testUtils.SemesterRepo
+
+	testUtils.ClassRepo.MockSelectAll = func(publishedOnly bool) ([]domains.Class, error) {
+		return createMockClasses(1), nil
+	}
+	repos.ClassRepo = &testUtils.ClassRepo
+
+	// Create new HTTP request to endpoint
+	recorder := testUtils.SendHttpRequest(t, http.MethodGet, "/api/classesbysemesters", nil)
+
+	// Validate results
+	assert.EqualValues(t, http.StatusOK, recorder.Code)
+	var results []domains.ProgramClassesBySemester
+	if err := json.Unmarshal(recorder.Body.Bytes(), &results); err != nil {
+		t.Errorf("unexpected error: %v\n", err)
+	}
+	assert.EqualValues(t, "2020_spring", results[0].Semester.SemesterId)
+	assert.EqualValues(t, "program1", results[0].ProgramClasses[0].ProgramObj.ProgramId)
+	assert.EqualValues(t, "program1_2020_spring_class1", results[0].ProgramClasses[0].Classes[0].ClassId)
+}
+
+func TestOneSemesterOneProgramOneClass_Failure(t *testing.T) {
```

Comment:
NoClasses?

Reply:
Woops, which you already did here (see below). If there's something specific that you're testing here that's NOT in the other test, you should keep both. However, if there isn't anything different between these two tests, just keep one of them and reduce test redundancy.

Reply:
I think here I tested what happens if one of the repo functions returns an error (which would fail)